src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -3352,6 +3542,136 @@ static class C
+            C.R(() => C.M(a));
+
+            static class C
+            {
+                public static void R(Expression<Action> e) => e.Compile()();
+                public static void M(object[] x) => Console.Write(" a" + x[0]);
+                public static void M(ReadOnlySpan<object> x) => Console.Write(" r" + x[0]);
+            }
+            """;
+        var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular12);
+        CompileAndVerify(comp, expectedOutput: "aa").VerifyDiagnostics();
+
+        var expectedOutput = "ra";
+
+        comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.RegularNext);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+
+        comp = CreateCompilationWithSpan(source);
+        CompileAndVerify(comp, expectedOutput: expectedOutput).VerifyDiagnostics();
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void OverloadResolution_ReadOnlySpanVsArray_ExpressionTree_03(LanguageVersion langVersion)
+    {
+        var source = """
+            using System;
+            using System.Linq.Expressions;
+
+            var a = new object[] { "a" };
+            C.R(() => C.M(a));
+
+            static class C
+            {
+                public static void R(Expression<Action> e) => e.Compile()();
+                public static void M(object[] x) => Console.Write(" a" + x[0]);
+                public static void M(ReadOnlySpan<object> x) => Console.Write(" r" + x[0]);
+            }
+            """;
+        var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion));
+        CompileAndVerify(comp, expectedOutput: "aa").VerifyDiagnostics();
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void OverloadResolution_ReadOnlySpanVsArray_ExpressionTree_04(LanguageVersion langVersion)
+    {
+        var source = """
+            using System;
+            using System.Linq.Expressions;
+            
+            C.R(() => C.M(null));
+            C.R(() => C.M(default));
+            
+            static class C
+            {
+                public static void R(Expression<Action> e) => e.Compile()();
+                public static void M(object[] x) => Console.Write(" 1" + (x?[0] ?? "null"));
+                public static void M(ReadOnlySpan<object> x) => Console.Write(" 2" + x.Length);
+            }
+            """;
+        var comp = CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion));
+        CompileAndVerify(comp, expectedOutput: "1null 1null").VerifyDiagnostics();
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void OverloadResolution_ReadOnlySpanVsArray_ExpressionTree_05(LanguageVersion langVersion)
+    {
+        var source = """
+            using System;
+            using System.Linq.Expressions;
+
+            C.R(() => C.M(default(object[])));
```

Comment:
>C.R(() => C.M(default(object[]))); [](http://example.com/codeflow?start=12&length=34)

Minor: Consider adding this case to the previous test rather than creating a separate test.